src/com/owncloud/android/utils/ErrorMessageAdapter.java

```diff
@@ -173,7 +173,11 @@ public static String getErrorCauseMessage(RemoteOperationResult result,
             }
         } else if (operation instanceof CreateShareViaLinkOperation ||
                     operation instanceof CreateShareWithShareeOperation) {
-            if (result.getCode() == ResultCode.SHARE_NOT_FOUND)  {      // Error --> SHARE_NOT_FOUND
+
```

Comment:
Unshare and createShare message calculation is basically the same code except for the chosen messages of course.

Would it make sense to put the message ids in a collection for example a hashmap with a key consisting of the message-id (maybe shortened and an key for the kind of operation.)? 
--> This way the code block could be shortened quite a bit :)